### PR TITLE
better error types for many of the STS rules

### PR DESCRIPTION
**Pull request by JaredCorduan**
I've made a bunch of the STS errors more descriptive.  The main things left to add are 1) `DELEG` failures, 2) maybe add more to the OCert signature failure (had some trouble with the show instance), and 3) multisig errors.

The `Delpl` rule has three errors that seemed to be totally unused:
```
ScriptNotInWitnessDELPL	
ScriptHashNotMatchDELPL	
ScriptDoesNotValidateDELPL
```
I have deleted these three error types.

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Bbody.hs [hunk 1]
```diff
@@ -66,8 +67,8 @@ instance
   type BaseM (BBODY crypto) = ShelleyBase
 
   data PredicateFailure (BBODY crypto)
-    = WrongBlockBodySizeBBODY
-    | InvalidBodyHashBBODY
+    = WrongBlockBodySizeBBODY Int Natural
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Bbody.hs]: Seems odd to report these in different types!

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Newpp.hs [hunk 2]
```diff
@@ -52,7 +52,7 @@ instance STS (NEWPP crypto) where
   type Environment (NEWPP crypto) = NewppEnv crypto
   type BaseM (NEWPP crypto) = ShelleyBase
   data PredicateFailure (NEWPP crypto)
-    = UnexpectedDepositPot
+    = UnexpectedDepositPot Coin Coin
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Newpp.hs:55]: Maybe nice to add some more doc indicating what these fields represent

**Review by nc6 — APPROVED** [review]
Massive improvement! Maybe some additional comments would also be useful to help interpret these

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Newpp.hs (hunk 2 above)
Comment (JaredCorduan) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Newpp.hs:55]: this is a great suggestion! if it sounds good to you, I'll add comments to all the STS errors in a follow up PR (and try to get the last few remaining ones that are still lacking proper details fleshed out).

**Pull request merged**